src/components/code-snippet/code-snippet.js

```diff
@@ -0,0 +1,77 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import handles from '../../globals/js/mixins/handles';
+import on from '../../globals/js/misc/on';
+
+class CodeSnippet extends mixin(createComponent, initComponentBySearch, handles) {
+  /**
+   * CodeSnippet UI.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @extends Handles
+   * @param {HTMLElement} element The element working as a CodeSnippet UI.
+   */
+
+  constructor(element, options) {
+    super(element, options);
+    this.manage(
+      on(this.element, 'click', event => {
+        this._handleClick(event);
+      })
+    );
+
+    this._initCodeSnippet();
+  }
+
+  _handleClick() {
+    const expandBtn = this.element.querySelector(this.options.classExpandText);
+    this.element.classList.toggle(this.options.classExpanded);
+
+    if (this.element.classList.contains(this.options.classExpanded)) {
+      expandBtn.textContent = expandBtn.getAttribute(this.options.showLessText);
+    } else {
+      expandBtn.textContent = expandBtn.getAttribute(this.options.showMoreText);
+    }
+  }
+
+  _initCodeSnippet() {
+    const expandBtn = this.element.querySelector(this.options.classExpandText);
+
+    expandBtn.textContent = expandBtn.getAttribute(this.options.showMoreText);
```

Comment:
Let’s put a guard against NPE (when `expandBtn` does not exist), by checking the existence here.

Reply:
For better explanation - Here's where the idea came from: https://github.com/carbon-design-system/carbon-components/pull/12